### ISSUE TITLE
Relay Rogue campaign_signup_post type to cio events to indicate kind of member action

### ISSUE THREAD

**Issue by sergii-tkachenko** [DoSomething/blink#125]
## TL;DR

**Sergii Tkachenko:** So, to summarize: "classic" Phoenix report back _item_* will become `campaign_signup_post` with `type: "action"`. And, for example, DACA call will be the same `campaign_signup_post`, but with type: `"call"`.

**Shae Smith:** yea that's about right. How we end up actually defining/counting a "reportback" will probably change in the coming months, but that is the idea.

_* - Added per Shae's clarification_

## Long Slack chat version:

[10:21 AM] 
Sergii Tkachenko Hey Shae, good morning!

[10:21 AM] 
Do you have time for a quick chat about setting reportback event type to cio signup post events?

[10:23 AM] 
So the idea is to assume that now all signups posts are reportbacks

[10:24 AM] 
so we'll add a type attribute to event arguments and set it to reportback

[10:24 AM] 
however, real event type is still stay campaign_signup_post

[10:24 AM] 
example

[10:25 AM] 
Sergii Tkachenko uploaded this image: Pasted image at 2017-09-22, 10:25 AM
Add Comment

[10:25 AM] 
Sergii Tkachenko
activity type (see col 3) will stay the same

[10:25 AM] 
and I'll extend actual data attributes with `type: "reportback",`

[10:26 AM] 
do you think this could be issue with current Rogue events implementation?

[10:26 AM] 
Shae Smith I see, thank you for the explanation.

[10:26 AM] 
> So the idea is to assume that now all signups posts are reportbacks

[10:27 AM] 
This is what is a little confusing- We tried in Rogue, to not assume that every post is a reportback

[10:27 AM] 
In fact, as we look towards multi-action campaigns, we really don't want to assume everything is a reportback

[10:28 AM] 
a Rogue `Post` paralleles what a `ReportbackItem` was in Phoenix-Ashes

[10:28 AM] 
and we say that a user has a "Reportback" if they have at least one of them

[10:28 AM] 
So if a user has 5 posts under a signup, they still only have one reportback

[10:30 AM] 
If your event classification works for C.io, then that is great, but just want to be clear about how were are defining stuff so that you all can be on the same page before segmenting things.

[10:31 AM] 
And some forward thinking- We might be including Post types in the `Post` model so that we can determine if a Post ties to different actions on a campaign

[10:31 AM] 
but that is to come

[10:32 AM] 
Sergii Tkachenko Ah, I see. Would it be less confusing to set the typo to `reportback_item` instead?

[10:32 AM] 
> And some forward thinking- We might be including Post types in the `Post` model so that we can determine if a Post ties to different actions on a campaign

[10:32 AM] 
yes, that would be perfect solution

[10:32 AM] 
Shae Smith
I would rather us also get rid of the "Reportback" terminology

[10:32 AM] 
maybe `action_item`?

[10:33 AM] 
or just `action`

[10:33 AM] 
Sergii Tkachenko yes, sound great

[10:33 AM] 
I like `action`

[10:34 AM] 
More forward thinking: could you please think of few examples when signup post events are not `actions`?

[10:34 AM] 
Insta post?

[10:35 AM] 
To clarify: trying to think what different post types could be on Post `model`

[10:36 AM] 
Shae Smith
sure!

[10:36 AM] 
so for example. There is a DACA campaign, and each day we give users new actions to take

[10:37 AM] 
a campaign lead would add actions, could be tweets, call your senator, IG posts, etc

[10:37 AM] 
and we would ingest those actions (somehow lol) as `Posts` (we could rethink this terminolgy in Rogue)

[10:39 AM] 
So a Post object could look something like this

[10:46 AM] 
Sergii Tkachenko Great example! Thanks.

[10:47 AM] 
Shae Smith
sorry my computer is being weird

[10:47 AM] 
but the idea is we want to leave it open so that a campaign lead could give users multple things to do

[10:47 AM] 
 ```
{
    id: "1",
    signup_id: "5970",
    campaign_id: "1603",
    northstar_id: "laksjdlfjadslf", 
    url: "http://www.url.com", // could be a link to a playlist, youtube video, image
    status: "pending",
    source: "phoenix-web",
    remote_addr: ""
    action_bucket: 1 // Each day would be a new "bucket", One campaign could have multiple buckets and each bucket could have multiple actions we ask the user to do. 
    action_type: "call", // The actual type of action "tweet", "call", "IG",etc 
    action_completes: true, // If the action completes the bucket or not. 
}
```
(edited)

[10:48 AM] 
none of the teminolgy has been solidified, and I don't even know if all of this is going to happen

[10:48 AM] 
but that is the idea

[10:48 AM] 
But i think considering each post and "action" for now is safe

[10:49 AM] 
Sergii Tkachenko So to summarize: "classic" Phoenix report back will become `campaign_signup_post` with  `type: "action"`. And, for example, DACA call will be the same `campaign_signup_post`, but with `type: "call"`. (edited)

[10:49 AM] 
Sounds about right?

[10:51 AM] 
Shae Smith yea that's about right. How we end up actually defining/counting a "reportback" will probably change in the coming months, but that is the idea.

**Comment by sbsmith86** [comment]
> So to summarize: "classic" Phoenix report back will become campaign_signup_post with type: "action". And, for example, DACA call will be the same campaign_signup_post, but with type: "call". (edited)

I should have caught this, but to be a little bit more clear: 

A "classic" Phoenix reportback **_item_** will become `campaign_signup_post` in c.io with type: "action". And, for example, DACA call will be the same campaign_signup_post, but with type: "call". (edited)

**Comment by sergii-tkachenko** [comment]
Yes, good 👁, Shae. Thank for clarifying!

**Comment by sergii-tkachenko** [comment]
@rapala61 This looks important. Could you reassign this to you?

**Comment by rapala61** [comment]
@sbsmith86 Found this as I'm getting ready to dive into Rogue changes in Blink. All C.io `campaign_signup_post`  are being created with a type of `photo`: https://github.com/DoSomething/blink/blob/master/src/messages/CampaignSignupPostMessage.js#L68. Seems you and Sergii talked about this a while back.

**Comment by sbsmith86** [comment]
@rapala61 yep! However that was implemented _before_ we started moving into the multiple action world. So before, everything was of `type = photo`. But we are starting to ingest different posts like `type=voter-reg` so we need to be able to send that to you.

**Comment by rapala61** [comment]
Got it! Thanks @sbsmith86. If you have a few minutes, I would love for you to take a glimpse at this Pivotal task I'm working on https://www.pivotaltracker.com/story/show/154736700. I updated the card with a few questions in regards to the changes I'm about to implement in the `user-signup-post` Blink route. We can also have a quick Slack call later if helps with being on the same page!

**Comment by sbsmith86** [comment]
@rapala61 I don't seem to have access to comment on the pivotal card. Do I have to be added to a team or something?

**Comment by rapala61** [comment]
I'll ask @justkika to add you as a member. I think that is the reason why you can't comment!
